test/functional/holds_controller_test.rb

```diff
@@ -24,56 +28,88 @@ class HoldsControllerTest < ActionController::TestCase
     # After cancellation of hold, status changed from "new" to "cancelled"
     assert_equal(resp_hold_obj.status, "cancelled")
     # Before cancellation of hold teacher-set available_copies count is 2.
-    # After cancellation of hold teacher-set available_copies count is 4.
+    # After cancellation of hold teacher-set available_copies count is 3.
     assert_equal(expected_ts_available_copies, TeacherSet.find(resp_hold_obj.teacher_set_id).available_copies)
     assert_response :redirect
     assert_equal("Your order was successfully updated.", flash[:notice])
   end
 
   test "test update method with empty holds" do
     sign_in @user
-    post :update, params: { id: @hold4.access_key, hold_change: {"comment" => "qqq", "status" => "cancelled"}, hold: {status: "MyText"} }
+    es_doc = {"_index" => "teacherset", "_type" => "teacherset", 
+              "_id" => @hold4.teacher_set.id, "_version" => 11, "result" => "updated", 
+              "_shards" => {"total" => 0, "successful" => 1, "failed" => 0}}
+
+    TeacherSet.stub_any_instance :update_teacher_set_availability_in_elastic_search, es_doc do
+      post :update, params: { id: @hold4.access_key, hold_change: {"comment" => "qqq", "status" => "cancelled"}, hold: {status: "MyText"} }
+    end
     assert_nil(flash[:notice])
   end
 
   test "Should fail hold cancelaltion" do
-    sign_in @user
     expected_ts_available_copies = 2
+    sign_in @user
+    resp = nil
+    es_doc = {"_index" => "teacherset", "_type" => "teacherset", 
+              "_id" => @hold2.teacher_set.id, "_version" => 11, "result" => "updated", 
+              "_shards" => {"total" => 0, "successful" => 1, "failed" => 0}}
 
-    resp = post :update, params: { id: @hold1.access_key, hold_change: {"comment" => "qqq", "status" => "new"}, hold: {status: "new"} }
-    resp_hold_obj = resp.request.env["action_controller.instance"].current_user.holds.find(@hold1.id)
+    # Teacher_set available_copies before cancellation of hold.
+    assert_equal(2, @hold2.teacher_set.available_copies)
+    
+    TeacherSet.stub_any_instance :update_teacher_set_availability_in_elastic_search, es_doc do
+      resp = post :update, params: { id: @hold2.access_key, hold_change: {"comment" => "qqq", "status" => "new"}, hold: {status: "new"} }
+    end
+    resp_hold_obj = resp.request.env["action_controller.instance"].current_user.holds.find(@hold2.id)
     # Hold status not changed.
-    assert_equal(resp_hold_obj.status, "new")
     # teacher-set available_copies also not changed, because hold is not cancelled.
+    assert_equal(resp_hold_obj.status, "new")
     assert_equal(expected_ts_available_copies, TeacherSet.find(resp_hold_obj.teacher_set_id).available_copies)
     assert_response :redirect
   end
 
   test "create hold and update teacher-set available_copies" do
-    expected_ts_available_copies = 0
+    expected_ts_available_copies = 1
```

Comment:
Please don't use `expected_ts_available_copies` here — just write `1` in the `assert_equals` statement below.